crates/bitwarden-crypto/src/store/context.rs

```diff
@@ -323,16 +512,25 @@ impl<Ids: KeyIds> KeyStoreContext<'_, Ids> {
         self.add_local_symmetric_key(SymmetricCryptoKey::make_aes256_cbc_hmac_key())
     }
 
-    /// Generate a new random xchacha20-poly1305 symmetric key and store it in the context
-    #[cfg(test)]
-    pub(crate) fn make_cose_symmetric_key(
+    /// Generate a new symmetric encryption key using the specified algorithm and store it in the
+    /// context as a local key
+    pub fn make_symmetric_key(&mut self, algorithm: SymmetricKeyAlgorithm) -> Ids::Symmetric {
```

Comment:
Not necesary as part of this PR, but now that we have this `make_symmetric_key` here, and the `move_symmetric_key` functions, we should audit all the uses of the deprecated `set_symmetric_key` and replace them if possible.

I know there are a lot of tests of the form:
```rust
        let symm_key = SymmetricCryptoKey::make_aes256_cbc_hmac_key();

        #[allow(deprecated)]
        store
            .context_mut()
            .set_symmetric_key(TestSymmKey::A(0), symm_key.clone())
            .unwrap();
```

that could be simplified with these new functions.

Reply:
I migrated a bunch of these now. `create_test_crypto_with_user_key` is not migrated yet, as this uses some static test vectors sometimes.